staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -969,3 +981,77 @@ type ResourceClaimTemplateList struct {
 	// Items is the list of resource claim templates.
 	Items []ResourceClaimTemplate `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+	// Driver specifies the name of the DRA driver whose kubelet
+	// plugin should be invoked to process the allocation once the claim is
+	// needed on a node.
+	//
+	// Must be a DNS subdomain and should end with a DNS domain owned by the
+	// vendor of the driver.
+	//
+	// +required
+	Driver string `json:"driver" protobuf:"bytes,1,rep,name=driver"`
+
+	// This name together with the driver name and the device name field
+	// identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+	//
+	// Must not be longer than 253 characters and may contain one or more
+	// DNS sub-domains separated by slashes.
+	//
+	// +required
+	Pool string `json:"pool" protobuf:"bytes,2,rep,name=pool"`
+
+	// Device references one device instance via its name in the driver's
+	// resource pool. It must be a DNS label.
+	//
+	// +required
+	Device string `json:"device" protobuf:"bytes,3,rep,name=device"`
+
+	// Conditions contains the latest observation of the device's state.
+	// If the device has been configured according to the class and claim
+	// config references, the `Ready` condition should be True.
+	//
+	// +optional
+	// +listType=atomic
+	Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,4,opt,name=conditions"`
+
+	// Data contains arbitrary driver-specific data.
```

Comment:
We need to spec a max size - @pohly did we do that for other RawExtension?

Reply:
Hmm, looks like we didn't. I'm torn between letting https://github.com/kubernetes/kubernetes/pull/127511 merge without it and then quickly following up with a fix and delaying that PR. Either way, I'll work on it now.

Reply:
Let's merge the beta promotion first as soon as possible (waiting for SIG Autoscaling) because it allows others (okay, primarily this PR here) to resolve conflicts.

Here's the [length limit PR](https://github.com/kubernetes/kubernetes/pull/128601).

Reply:
@LionelJouin: you own validation will be simpler if enforcing the limit from day one.

Reply:
Once those PRs are merged I will rebase and set this limitation

Reply:
@LionelJouin https://github.com/kubernetes/kubernetes/pull/127511 merged, you should rebase now on top of that and set the limitation on size as in https://github.com/kubernetes/kubernetes/pull/128601

Reply:
Now the rebase is done with the size limitation